### ISSUE TITLE
High CPU usage in watcher service

### ISSUE THREAD

**Issue by aeschli** [microsoft/vscode-remote-release#4649]
I have a similar issue.
Using WSL2 with debian buster.

![image](https://user-images.githubusercontent.com/5438762/110669857-8b379b00-81ab-11eb-880d-d1a3c9599cc5.png)

Comparing with the windows task manager:

![image](https://user-images.githubusercontent.com/5438762/110670306-026d2f00-81ac-11eb-9f63-130c80be2e87.png)

The drop in the graphs is because I closed vscode to compare the average CPU usage.
While vscode is open, CPU usage in all cores is 60%~90%.

It doesn't seem to be extension related, because it isn't the `extensionHost` that is using CPU.

I'm using Windows 10 Pro 1909 build 18363.1379

_Originally posted by @Hellysonrp in https://github.com/microsoft/vscode-remote-release/issues/3171#issuecomment-795783235_

**Comment by aeschli** [comment]
@Hellysonrp Do you have multiple WSL windows open? Are they on folders or workspaces in `/mnt/...`?

**Comment by Hellysonrp** [comment]
@aeschli 
> Do you have multiple WSL windows open?

No, just one.

> Are they on folders or workspaces in `/mnt/...`?

Yes, it's a workspace with ~10 projects, with golang and protobuf files.
All my projects are in the `C:\Users\Hellysonrp\Documents\GitHub` folder (or `/mnt/c/...`).

**Comment by martinbertolino** [comment]
FWIW, I had noticed the same thing on a new development machine I just got.  A node process associated with VS Code 1.54.2 in WSL 1 was steadily consuming 25% of my CPU.  Upgrading to WSL 2 did not mate any difference.  Same behavior with Ubuntu 18.04 and with 20.04.  Eventually I paid attention to a popup from VS Code right at startup about not having my code/projects under the /mnt/c/ tree.  Once I moved my code under the /home/${myuser}/code/project .... so I presume a Linux filesystem rather than one on the Windows side, the CPU consumption decreased significantly, and it now hovers between 6% and 9% when not doing anything.  Fan is not running constantly anymore.  Hope this helps.

**Comment by aeschli** [comment]
I suspect this was caused by the change to use polling for the Windows file system mounts. I have now disabled this by default. Published ms-vscode-remote.remote-wsl@0.54.5
Can you test if this has improved the situation?

- verify that the latest version of the extension is installed
- close all VSCode windows
- wait 10 seconds . There should be no more `code` process running in WSL
- restart

**Comment by Hellysonrp** [comment]
Yes, it has improved a lot.
CPU usage with vscode open is now ~15% instead of ~60%.

**Comment by josalhor** [comment]
I am also impacted by this. May I ask what the patch does? Does it simply not refresh the windows file system on /mnt/ ?

**Comment by aeschli** [comment]
Yes, by switching back to regular file watching (without polling), we are back to missing file events when working on `/mnt/c/..`.
Its a known WSL2 issue that does not only impact VSCode. The recommendation is to locate projects on the Linux file system.

**Comment by singhkays** [comment]
Noticing a similar issue on a remote Linux (Oracle Linux/Centos) based host - 1 Core/7 GB memory

![image](https://user-images.githubusercontent.com/6164178/112207677-a905f680-8bd4-11eb-9dbb-7287137d89db.png)

## VSCode version
Version: 1.55.0-insider
Commit: a6f6b4aa8e3a688d9a5cbcd5df14af511ce2bd8e
Date: 2021-03-23T05:26:59.637Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Darwin arm64 20.3.0

**Comment by BrianNestrick** [comment]
I have been running into the same problem with version 0.58.2/WSL2/Ubuntu.  Is the recommendation still to revert the extension back to `ms-vscode-remote.remote-wsl@0.54.5`?

**Comment by aeschli** [comment]
Please use the latest version. The performance issue cause by polling has been fixed in the meantime.
Can you run `htop` in WSL to find out which process is using the CPU? Press 'upper case P` to sort by CPU usage and use cursor right to see the full command line.
